src/core/Akka/Serialization/Serialization.cs

```diff
@@ -145,14 +146,15 @@ private Serializer GetSerializerByName(string name)
         public ActorSystem System { get; }
 
         /// <summary>
-        /// TBD
+        /// Adds the serializer to the internal state of the serialization subsystem
         /// </summary>
-        /// <param name="serializer">TBD</param>
-        /// <returns>TBD</returns>
+        /// <param name="name">Configuration name of the serializer</param>
+        /// <param name="serializer">Serializer instance</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void AddSerializer(Serializer serializer)
+        public void AddSerializer(string name, Serializer serializer)
```

Comment:
This is a breaking API change since we're modifying a public method. IMHO, add a new overload and mark the old one as deprecated. That way we won't have binary compatibility issues on upgrade.

Reply:
You are correct. However seeing as its being aggressively inlined. Im wondering why it was public in the first place. But ok.

Reply:
@Danthar 
> Im wondering why it was public in the first place. 

I'm right there with you.